sky/packages/sky/lib/rendering/object.dart

```diff
@@ -294,31 +360,45 @@ typedef void RenderObjectVisitor(RenderObject child);
 typedef void LayoutCallback(Constraints constraints);
 typedef double DimensionCallback(Constraints constraints);
 
+/// An object in the render tree
+///
+/// Render objects have a reference to their parent but do not commit to a model
+/// for their children.
 abstract class RenderObject extends AbstractNode implements HitTestTarget {
 
   // LAYOUT
 
-  // parentData is only for use by the RenderObject that actually lays this
-  // node out, and any other nodes who happen to know exactly what
-  // kind of node that is.
+  /// Data for use by the parent render object
+  ///
+  /// The parent data is used by the render object that lays out this object
+  /// (typically this object's parent in the render tree) to store information
+  /// relevant to itself and to any other nodes who happen to know exactly what
+  /// the data means. The parent data is opaque to the child.
```

Comment:
Add the following information:
 - that the parentData object must inherit from ParentData (despite being typed as `dynamic`)
 - that the parentData field must not be directly set, except by calling setupParentData() on the parent node
 - that the parentData can be set before the child is added to the parent, by calling setupParentData on the future parent node
 - that the parent has responsibility over this field
 - that whether the child is allowed to access this field depends entirely on the convention defined by the ParentData subclass. (For example, BoxParentData is defined as being opaque to the child, so subclasses of RenderBox must not access their own parentData field.)